contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2509,3 +2510,137 @@ get_owner_of_schema(const char *schema)
 
 	return result;
 }
+
+/* Find namespace oid of a procedure based on proc name. */
+static Oid
+get_proc_namespace_oid(char **proc_name, char *curr_db)
+{
+	char *physical_sch_name;
+	char *db_name;
+	char *schema_name;
+	char	  **splited_object_name;
+	Oid obj_schema_oid = InvalidOid;
+
+	/* Resolve the three part name. */
+	splited_object_name = split_object_name(*proc_name);
+	db_name = splited_object_name[1];
+	schema_name = splited_object_name[2];
+	*proc_name = splited_object_name[3];
+
+	if (!strcmp(db_name, ""))
+		db_name = curr_db;
+
+	if (!strcmp(schema_name, ""))
+	{
+		/* Find the default schema for current user and get physical schema name */
+		const char *user = get_user_for_database(db_name);
+		schema_name = get_authid_user_ext_schema_name((const char *) db_name, user);
+	}
+	else
+	{
+		schema_name = downcase_truncate_identifier(schema_name, strlen(schema_name), true);
+	}
+
+	/* Get physical schema name from logical schema name */
+	physical_sch_name = get_physical_schema_name(db_name, schema_name);
+	/* Get namespace oid from physical schema name */
+	obj_schema_oid = get_namespace_oid(physical_sch_name, false);
+
+	return obj_schema_oid;
+
+}
+
+static Oid
+get_proargtypes_oid(char *proname, Oid pronamespace, Oid user_id, char *targeted_arg_name)
+{
+	HeapTuple    tuple;
+	CatCList   *catlist;
+	Oid matched_type = InvalidOid;
+
+	targeted_arg_name = downcase_truncate_identifier(targeted_arg_name, strlen(targeted_arg_name), true);
+
+	/*
+	 * Search pg_proc for the procedure by name in the specified namespace and
+	 * return the argument type, if the targeted argument name matches
+	 */
+	catlist = SearchSysCacheList1(PROCNAMEARGSNSP, CStringGetDatum(proname));
```

Comment:
Can we directly filter with pronamespace since overloading is restricted?

Reply:
As per our offline discussion, this won't be possible, as PG allows overloading and to search by procedure name, and namespace we would also require arg list, which is not available in our case.